src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -153,7 +186,29 @@ private void EnsureAppDataSafety()
             if (_appFolderInfo.StartUpFolder.IsParentPath(_appFolderInfo.AppDataFolder) ||
                 _appFolderInfo.StartUpFolder.PathEquals(_appFolderInfo.AppDataFolder))
             {
-                throw new NotSupportedException("Update will cause AppData to be deleted, correct you configuration before proceeding");
+                throw new UpdateFailedException("Update will cause AppData to be deleted, correct you configuration before proceeding");
```

Comment:
"You sonarr configuration is being stored in Application folder which will cause data lost during the upgrade. Please remove any symlinks or redirects before trying again."